source/_posts/2017-06-14-Ansible-via-python.markdown

```diff
@@ -0,0 +1,188 @@
+# Ansible via Python ![alt text](https://cub.nobleprog.com/sites/hitramx/files/styles/height50_scale/public/category_image/cursos-de-ansible-en-mexico.png?itok=xPUrGNrA)
+
+**_Ansible is designed around the way people work and the way people work together_**
+
+## What is Ansible
+
+Ansible is simple IT engine for automation, it is designed for manage many
+systems, rather than just one at a time. Ansible automates cloud provisioning,
+configuration management, application deployment, intra-service
+orchestration, and many other IT operations. It is easy to deploy due to using
+no agents and no additional custom security infrastructure. We can define own
+configuration management in simple YAML language, which is named
+ansible-playbook. YAML is easier to read and write by humans than other common
+data formats line XML or JSON. Futhermore, most programming languages contain
+libraries which operate and work YAML.
+
+## Inventory
+
+Ansible works collaterally on many systems in your infrastructure, so it is
+important to specify a roster to keep `hosts`. This list is named `inventory`,
+which can be in one of many formats. For this example, the format is an
+`INI-like` and is saved in `/etc/ansible/hosts`.
+
+```INI
+mail.example.com
+
+[webservers]
+foo.example.com
+bar.example.com
+
+[dbservers]
+one.example.com
+two.example.com
+three.example.com
+```
+
+## Ansible Playbook
+
+Playbook is the way to direct systems, which is particularly powerful, compact
+and easy to read and write. As we said configuration multi-system management is
+formatted in YAML language. Ansible playbook consists of `plays`, which
+contain `hosts` we would like to manage, and `tasks` we want to perform.
+
+```yaml
+---
+- hosts: webservers
+  vars:
+    http_port: 80
+    max_clients: 200
+  remote_user: root
+  tasks:
+  - name: ensure apache is at the latest version
+    yum: name=httpd state=latest
+  - name: write the apache config file
+    template: src=/srv/httpd.j2 dest=/etc/httpd.conf
+    notify:
+    - restart apache
+  - name: ensure apache is running (and enable it at boot)
+    service: name=httpd state=started enabled=yes
+  handlers:
+    - name: restart apache
+      service: name=httpd state=restarted
+```
+
+### Ansible for Embedded Linux
+
+There is possibility to need build image with custom Linux-based system for
+embedded with Ansible. In addition, we would like to run ansible-playbook via
+python. It seems to be hard to implement. Nothing more simple! It is needed to
+add recipe to image with ansible from
+[Python Ansible package](https://github.com/OverC/meta-overc/blob/master/meta-cube/recipes-devtools/python/python-ansible_2.1.1.0.bb)
```

Comment:
@Macias118 please add some context. We cannot assume that Embedded Linux equals Yocto. Please be clear that you wrote about Yocto meta layer with support for Ansible.